pkg/controller/agent/reconcile.go

```diff
@@ -128,23 +134,36 @@ func reconcileDaemonSet(rp ReconciliationParams) (int32, int32, error) {
 	return reconciled.Status.NumberReady, reconciled.Status.DesiredNumberScheduled, nil
 }
 
+// ReconciliationParams are the parameters used during an Elastic Agent's reconciliation.
 type ReconciliationParams struct {
 	client      k8s.Client
 	agent       agentv1alpha1.Agent
 	podTemplate corev1.PodTemplateSpec
 }
 
-func updateStatus(params Params, ready, desired int32) error {
+// calculateStatus will calculate a new status from the state of the pods within the k8s cluster
+// and will return the new status, and any errors encountered.
+func calculateStatus(params *Params, ready, desired int32) (agentv1alpha1.AgentStatus, error) {
+	status := newStatus(params.Agent)
```

Comment:
nit: a new status has already been created at the beginning of the reconciliation loop and a pointer is available in the context of `reconcilePodVehicle`. Should we just use it instead of creating a new one and replacing the content ?

```suggestion
// calculateStatus updates the current status using the state of the pods within the k8s cluster.
func calculateStatus(params *Params, ready, desired int32, status *agentv1alpha1.AgentStatus) error {
```

Reply:
👍